lang/cs/Org.Apache.REEF.Driver/Bridge/DriverRestartHandlerWrapper.cs

```diff
@@ -43,7 +46,17 @@ internal sealed class DriverRestartHandlerWrapper : IObserver<StartTime>
 
         public void OnNext(StartTime startTime)
         {
-            _driverRestartedObserver.OnNext(new DriverRestarted(new DateTime(startTime.TimeStamp)));
+            if (_driverRestartedObserver is DefaultDriverRestartedHandler)
```

Comment:
This is because the DefaultDriverRestartedHandler throws, so if the user only binds DriverRestartHandler but not DriverRestartedHandler, it actually fails the job instead of the advertised behavior, which is to defer the behavior to DriverRestartHandler. I will add the else case.